### DISCUSSION TITLE
Counting included model doesn't work

### DISCUSSION THREAD

**Discussion by thoriqadillah** [sequelize/sequelize#15091] [Help & Questions]
## Issue Creation Checklist

- [X] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Issue Description
My query to count included model with `Model.findOne` doesn't work properly

### Additional context

Here is my query
```js
const sequelize = require('sequelize')

TrainingClass.findOne({ 
  where: { 
    id: req.id,
      training_id: req.trainingId
  },
  attributes: {
    include: [
      [sequelize.fn('COUNT', 'students.id'), 'totalStudents'],
    ]
  },
  include: [
    {
      model: TrainingClassSubject,
      as: 'subjects',
      attributes: ['trainer_id']
    },
    {
      model: TrainingClassStudent,
      as: 'students',
      attributes: ['student_id']
    }
  ],
  group: ['subjects.id', 'TrainingClass.id', 'students.id'],
})
```
And produces this result
```js
"data": {
    ...TrainingClassData
    "totalStudents": "1",
    "subjects": [
      {
        "trainer_id": "4"
      },
      {
        "trainer_id": "5"
      },
      {
        "trainer_id": "2"
      },
      {
        "trainer_id": "5"
      }
    ],
    "students": [
      {
        "student_id": "4"
      },
      {
        "student_id": "5"
      },
      {
        "student_id": "10"
      },
      {
        "student_id": "12"
      },
      {
        "student_id": "12"
      },
      {
        "student_id": "13"
      }
    ]
  }
```
I actually don't want to return the `student_id` in my actual case, it's just to give this question a context 
As you can see, the count fn result doesn't match with actual data if you count it manually. How can I resolve this?
Query log:
```
Executing (default): SELECT "TrainingClass"."id", "TrainingClass"."training_id", "TrainingClass"."slug", "TrainingClass"."name", "TrainingClass"."description", "TrainingClass"."type", "TrainingClass"."start_register", "TrainingClass"."end_register", "TrainingClass"."start_class", "TrainingClass"."end_class", "TrainingClass"."price", "TrainingClass"."discount", "TrainingClass"."created_at" AS "createdAt", "TrainingClass"."updated_at" AS "updatedAt", "TrainingClass"."deleted_at" AS "deletedAt", COUNT("students"."id") AS "totalStudents", "subjects"."id" AS "subjects.id", "subjects"."trainer_id" AS "subjects.trainer_id", "students"."id" AS "students.id", "students"."student_id" AS "students.student_id" FROM "public"."training_classes" AS "TrainingClass" LEFT OUTER JOIN "public"."training_class_subjects" AS "subjects" ON "TrainingClass"."id" = "subjects"."training_class_id" AND ("subjects"."deleted_at" IS NULL) LEFT OUTER JOIN "public"."training_class_students" AS "students" ON "TrainingClass"."id" = "students"."training_class_id" AND ("students"."deleted_at" IS NULL) WHERE ("TrainingClass"."deleted_at" IS NULL AND ("TrainingClass"."id" = 2 AND "TrainingClass"."training_id" = 1)) GROUP BY "subjects"."id", "TrainingClass"."id", "students"."id";
```
PS: 
- Same thing happen with `[sequelize.fn('COUNT', sequelize.col('students.id')), 'totalStudents']`
- I already found this problem asked in [stackoverflow](https://stackoverflow.com/questions/52496842/sequelize-hasmany-associatedmodel-count-in-attributes-in-query-execution) but the query somehow doesn't work with me despite the same query.

### Is this issue dialect-specific?

- [ ] No. This issue is relevant to Sequelize as a whole.
- [ ] Yes. This issue only applies to the following dialect(s): XXX, YYY, ZZZ
- [X] I don't know.

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [X] No, I don't have the time and I wouldn't even know how to start.

**Comment by ephys** [comment]
You're grouping by `students.id`, counting `students.id` is going to return 1 for each group

**Comment by thoriqadillah** [comment]
So how do i return it the entire data count at once rather than summing one by one?
If i remove the `students.id` in group attribute, it will yell at me "students.id" must appear on group by or something
